src/Core/IdentityServer/ClientStore.cs

```diff
@@ -110,14 +121,70 @@ public async Task<Client> FindClientByIdAsync(string clientId)
+                        claims.AddRange(new List<ClientClaim>
+                        {
+                            new ClientClaim("premium", isPremium ? "true" : "false", ClaimValueTypes.Boolean),
+                            new ClientClaim(JwtClaimTypes.Email, user.Email),
+                            new ClientClaim(JwtClaimTypes.EmailVerified, user.EmailVerified ? "true" : "false",
+                                ClaimValueTypes.Boolean),
+                            new ClientClaim("sstamp", user.SecurityStamp)
+                        });
+
+                        if (!string.IsNullOrWhiteSpace(user.Name))
+                        {
+                            claims.Add(new ClientClaim(JwtClaimTypes.Name, user.Name));
+                        }
+
+                        // Orgs that this user belongs to
+                        var orgs = await _currentContext.OrganizationMembershipAsync(_organizationUserRepository, user.Id);
+                        if (orgs.Any())
+                        {
+                            foreach (var group in orgs.GroupBy(o => o.Type))
+                            {
+                                switch (group.Key)
+                                {
+                                    case Enums.OrganizationUserType.Owner:
+                                        foreach (var org in group)
+                                        {
+                                            claims.Add(new ClientClaim("orgowner", org.Id.ToString()));
+                                        }
+                                        break;
+                                    case Enums.OrganizationUserType.Admin:
+                                        foreach (var org in group)
+                                        {
+                                            claims.Add(new ClientClaim("orgadmin", org.Id.ToString()));
+                                        }
+                                        break;
+                                    case Enums.OrganizationUserType.Manager:
+                                        foreach (var org in group)
+                                        {
+                                            claims.Add(new ClientClaim("orgmanager", org.Id.ToString()));
+                                        }
+                                        break;
+                                    case Enums.OrganizationUserType.User:
+                                        foreach (var org in group)
+                                        {
+                                            claims.Add(new ClientClaim("orguser", org.Id.ToString()));
+                                        }
+                                        break;
+                                    default:
+                                        break;
+                                }
+                            }
```

Comment:
Is this all the same as what is in the profile service? Should we break this out into a helper so we are not repeating ourselves?

Reply:
I had thought about doing that, but there is one difference that complicates things: the `ProfileService` logic creates a `ICollection<Claim>` and the `Client().Claims` property here is expecting a `ICollection<ClientClaim>`. It doesn't even look like these two share a base class, so I'm not sure a helper would work.

Reply:
Could you just return a basic <string, string> dictionary or KVP or something and translate that response to the Claim and ClientClaim? If the string value is "true" or "false" mark it as Boolean type. All others seem to be just regular <string, string>

Reply:
Good idea! Implemented as described.